internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,713 @@
+						AtLeastOneOf: []string{
+							"platform.0.edge_device_id",
+							"platform.0.stack_hci_cluster_id",
+							"platform.0.arc_kubernetes_cluster_id",
+							"platform.0.custom_location_id",
+						},
+					},
+
+					"type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(packetcorecontrolplane.PlatformTypeAKSNegativeHCI),
+							string(packetcorecontrolplane.PlatformTypeThreePNegativeAZURENegativeSTACKNegativeHCI),
+							"BaseVM", // tracked on https://github.com/Azure/azure-rest-api-specs/issues/23243
+						}, false),
+					},
+				},
+			},
+		},
+
+		"identity": commonschema.UserAssignedIdentityOptional(),
+
+		"interoperability_settings_json": {
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			ValidateFunc:     validation.StringIsJSON,
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"local_diagnostics_access": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 1,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"authentication_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(packetcorecontrolplane.AuthenticationTypeAAD),
+							string(packetcorecontrolplane.AuthenticationTypePassword),
+						}, false),
+					},
+					"https_server_certificate_url": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.Tags(),
+
+		"version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r PacketCoreControlPlaneResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r PacketCoreControlPlaneResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model PacketCoreControlPlaneModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MobileNetwork.PacketCoreControlPlaneClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := packetcorecontrolplane.NewPacketCoreControlPlaneID(subscriptionId, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			identityValue, err := expandMobileNetworkLegacyToUserAssignedIdentity(model.Identity)
```

Comment:
We should be able to use the expand function from the identity package here?

Reply:
(as same as above)